### PR TITLE
*: refactor Handle in executor package

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve?

Problem Summary:
refacor Handle in executor package for cluster index.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
- add HandleMap.
- refacor Handle in executor package for cluster index.

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: kv/key.go [hunk 1]
```diff
@@ -164,6 +164,9 @@ func (ih IntHandle) IntValue() int64 {
 
 // Next implements the Handle interface.
 func (ih IntHandle) Next() Handle {
+	//if ih.IsMax() {
```
Comment (cfzjywxk) [inline kv/key.go:167]: Should uncomment this?

File: executor/index_merge_reader.go [hunk 2]
```diff
@@ -621,14 +620,10 @@ func (w *partialIndexWorker) extractTaskHandles(ctx context.Context, chk *chunk.
...
 }
 
-func (w *partialIndexWorker) buildTableTask(handles []int64, retChk *chunk.Chunk) *lookupTableTask {
-	var indexOrder map[int64]int
-	var duplicatedIndexOrder map[int64]int
+func (w *partialIndexWorker) buildTableTask(handles []kv.Handle, retChk *chunk.Chunk) *lookupTableTask {
 	task := &lookupTableTask{
-		handles:              handles,
-		indexOrder:           indexOrder,
-		duplicatedIndexOrder: duplicatedIndexOrder,
```
Comment (cfzjywxk) [inline executor/index_merge_reader.go:630]: Should initialize these two fields ?

Comment (coocood) [reply executor/index_merge_reader.go:630]: The original code just declared but not initialized the two maps.
And I checked that IndexMergeReader doesn't care about the order of the rows, so the two maps are not used.

**Comment by cfzjywxk** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**